velox/core/QueryCtx.cpp

```diff
@@ -81,6 +101,15 @@ QueryCtx::QueryCtx(
   return fmt::format("query.{}.{}", queryId.c_str(), seqNum++);
 }
 
+void QueryCtx::maybeSetReclaimer() {
+  VELOX_CHECK_NOT_NULL(pool_);
+  if (pool_->reclaimer() != nullptr) {
+    return;
+  }
+  underArbitration_ = false;
```

Comment:
In what scenario will pool be non null and underArbitration_ be true when this method is called?

Reply:
underArbitration_ won't be true.